### PR TITLE
Serialize Plan's roles and base_framework fields

### PR DESCRIPTION
**Pull request by vvmnnnkv**
## Description
This PR complements addition on PlanTranslatorTfJs and multiple roles in Plan (#3371).
It updates Plan serialization methods to serialize new fields (`base_framework` and `roles`).

## Affected Dependencies
Uses syft-proto 0.4.9.

## How has this been tested?
Extended existing serde & protobuf unit tests to check new fields.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: syft/execution/plan.py [hunk 1]
```diff
@@ -490,8 +491,9 @@ def call_wrapped_framework(role):
 
     def add_translation(self, plan_translator: "AbstractPlanTranslator"):
         role = plan_translator(self).translate()
-        self.roles[plan_translator.framework] = role
-        return role
+        if isinstance(role, Role):
```
Comment (gmuraru) [inline syft/execution/plan.py:494]: Q: When is the case when ```role``` is not a ```Role```?

File: syft/execution/translation/threepio.py [hunk 2]
```diff
@@ -31,19 +31,18 @@ def translate_action(self, action: Action, to_framework: str) -> Action:
...
     def translate_framework(self, to_framework: str) -> Role:
         """Translates current plan's Role to specified framework"""
-        role = self.plan.role.copy()
-        plan = self.plan.copy()
+
         # Check to see if plan has been translated to this framework yet
-        if plan.roles.get(to_framework, None) is not None:
-            plan.default_framework = to_framework
-            return plan
+        if self.plan.roles.get(to_framework, None) is not None:
```
Comment (gmuraru) [inline syft/execution/translation/threepio.py:36]:
```suggestion
        if self.plan.roles.get(to_framework, None):
```

File: test/notebooks/test_notebooks.py [hunk 3]
```diff
@@ -149,7 +149,7 @@ def test_notebooks_advanced(isolated_filesystem, notebook):
     notebook = notebook.replace("examples/tutorials/", "")
     list_name = Path("examples/tutorials/") / notebook
     tested_notebooks.append(str(list_name))
-    res = pm.execute_notebook(notebook, "/dev/null", parameters={"epochs": 1}, timeout=300)
+    res = pm.execute_notebook(notebook, "/dev/null", parameters={"epochs": 1}, timeout=400)
```
Comment (gmuraru) [inline test/notebooks/test_notebooks.py:152]: Q: This is because the notebooks tests sometimes timeout, right?

File: syft/execution/plan.py (hunk 1 above)
Comment (vvmnnnkv) [reply syft/execution/plan.py:494]: Torchscript translator returns Plan: 
https://github.com/OpenMined/PySyft/blob/vvm/plan-roles-serialize/syft/execution/translation/torchscript.py#L50
It's different kind of translator that doesn't produce new "role" but sets torchscript property in Plan.
Well, I guess we'll eventually need to refactor to move that property to Role level.

File: test/notebooks/test_notebooks.py (hunk 3 above)
Comment (vvmnnnkv) [reply test/notebooks/test_notebooks.py:152]: Yes, specifically Encrypted Linear Regression tutorial, and I don't think this timeout is related to change in this PR, just sometimes it exceeded timeout (sometimes not).

**Pull request merged**